pulp_rpm/app/serializers/advisory.py

```diff
@@ -201,8 +201,13 @@ def validate(self, data):
         """
         update_record_data = dict()
         if "file" in data:
-            update_record_data.update(json.loads(data["file"].read()))
-            update_record_data.update(data)
+            try:
+                update_record_data.update(json.loads(data["file"].read()))
+                update_record_data.update(data)
+            except ValueError as e:
```

Comment:
when `data["file"]` is not a JSON, at the issue, it raised `UnicodeDecodeError`.
testing locally I got:
```python
In [1]: import json                                                                                                                                                                                                                           

In [2]: json.loads("a")                                                                                                                                                                                                                       
---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
<ipython-input-2-c0c0ca6f67c6> in <module>
----> 1 json.loads("a")

~/.pyenv/versions/3.8.0/lib/python3.8/json/__init__.py in loads(s, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    355             parse_int is None and parse_float is None and
    356             parse_constant is None and object_pairs_hook is None and not kw):
--> 357         return _default_decoder.decode(s)
    358     if cls is None:
    359         cls = JSONDecoder

~/.pyenv/versions/3.8.0/lib/python3.8/json/decoder.py in decode(self, s, _w)
    335 
    336         """
--> 337         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    338         end = _w(s, end).end()
    339         if end != len(s):

~/.pyenv/versions/3.8.0/lib/python3.8/json/decoder.py in raw_decode(self, s, idx)
    353             obj, end = self.scan_once(s, idx)
    354         except StopIteration as err:
--> 355             raise JSONDecodeError("Expecting value", s, err.value) from None
    356         return obj, end

JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Reply:
oh! it is a subclass!
today I learned!
thank you @ipanova

Reply:
While `JSONDecodeError` is a subclass of `ValueError`, I think it would be "cleaner" to catch the specific exception. Otherwise a `ValueError` coming from some different location could be hidden. Granted I don't see such a location in the _current_ code, I think it's a good idea to be as specific as possible.